src/main/java/org/broadinstitute/hellbender/utils/MannWhitneyU.java

```diff
@@ -534,10 +538,10 @@ public double permutationTest(final double[] series1, final double[] series2, fi
          * and dividing by the total count of everything in the histogram. Just using getCumulativeDistribution() gives
          * a p-value of 1 in the most extreme case which doesn't result in a usable z-score.
          */
-        double sumOfAllSmallerBins = histo.get(testStatU).getValue() / 2.0;
+        double sumOfAllSmallerBins = histo.get(FastMath.round(2 * testStatU)).getValue() / 2.0;
 
-        for (final Histogram.Bin<Double> bin : histo.values()) {
-            if (bin.getId() < testStatU) sumOfAllSmallerBins += bin.getValue();
+        for (final Histogram.Bin<Long> bin : histo.values()) {
+            if (bin.getId() < FastMath.round(2 * testStatU)) sumOfAllSmallerBins += bin.getValue();
```

Comment:
I'm probably missing something here, but can you explain why `bin.getValue()` is not divided by 2.0 here?

Reply:
Because the value is just the count, and we only modified the key.

Reply:
Ah yes, I got confused because the code above is purposefully taking "half of the count in the observed bin".

:+1: merge away